packages/override-tools/src/Cli.ts

```diff
@@ -0,0 +1,210 @@
+          version: {type: 'string', describe: 'React Native Version'},
+        }),
+      cmdArgv => validateManifest(cmdArgv.manifest, cmdArgv.version),
+    )
+    .epilogue(
+      'This tool allows maging JavaScript overrides for React Native Windows',
+    )
+    .demandCommand()
+    .recommendCommands()
+    .strict()
+    .showHelpOnFail(false)
+    .wrap(yargs.terminalWidth())
+    .version(false).argv;
+});
+
+/**
+ * Check that the manifest correctly describes files and is up to date
+ */
+async function validateManifest(manifestPath: string, version?: string) {
+  try {
+    await fs.promises.access(manifestPath);
+  } catch {
+    console.error(
+      chalk.red(`Could not find manifest file at path '${manifestPath}'`),
+    );
+    return;
+  }
+
+  const spinner = ora();
+
+  try {
+    spinner.start('Initializing local React Native repo');
+    const [reactRepo, ovrRepo] = await getFileRepos(manifestPath);
+    spinner.succeed();
+
+    let manifest: Manifest.Manifest;
+
+    try {
+      manifest = await Manifest.readFromFile(manifestPath);
+    } catch (ex) {
+      console.error(chalk.red('Could not parse manifest. Is it valid?'));
+      throw ex;
+    }
+
+    spinner.start('Verifying overrides are recorded and up to date');
+    const rnVersion = version || (await getInstalledRNVersion(manifestPath));
+    const validationErrors = await Manifest.validate(
+      manifest,
+      ovrRepo,
+      FileRepository.bindVersion(reactRepo, rnVersion),
+    );
+
+    if (validationErrors.length === 0) {
+      spinner.succeed();
+    } else {
+      spinner.fail();
+      printValidationErrors(validationErrors);
+      process.exitCode = 1;
+    }
+  } catch (ex) {
+    if (spinner.isSpinning) {
+      spinner.fail();
+    }
+    throw ex;
+  }
+}
+
+/**
+ * Create and initialize file repos
+ */
+async function getFileRepos(
+  manifestPath: string,
+): Promise<[GitReactFileRepository, OverrideFileRepositoryImpl]> {
+  const gitDir = os.tmpdir() + path.sep + 'react-native-windows-override-git';
+  const reactRepo = await GitReactFileRepository.createAndInit(gitDir);
+
+  const ovrPath = path.dirname(manifestPath);
+  const ovrFilter = /^.*\.(js|ts|jsx|tsx)$/;
+  const ovrRepo = new OverrideFileRepositoryImpl(ovrPath, ovrFilter);
+
+  return [reactRepo, ovrRepo];
+}
+
+/**
+ * Try to find the currently installed version of React Native by seaching for
+ * it's package.json.
+ */
+async function getInstalledRNVersion(manifestPath: string): Promise<string> {
+  const relativeRnPackage = 'node_modules\\react-native\\package.json';
+
+  let searchPath = path.resolve(manifestPath);
+  while (path.resolve(searchPath, '..') !== searchPath) {
+    try {
+      const rnPackage = path.join(searchPath, relativeRnPackage);
+      const packageJson = (await fs.promises.readFile(rnPackage)).toString();
+      const version = JSON.parse(packageJson).version;
+
+      if (!version) {
+        throw new Error();
```

Comment:
>Error [](start = 18, length = 5)

(nit) should have some error message

Reply:
This is being used strictly for control flow (I know, gross).

Catching exceptions/promise rejection is basically the only way to check if a file exists using non-deprecated functions in Node. We need a catch for control flow below a anyway, so we throw here to go to the next loop iteration.

Reply:
For more context, fs.access is the current preferred way to test that a file exists and it does so by throwing if the test fail. Node documentation reccomends just catching the exception from trying to read instead.

I really don't like the API, but I don't think there's a sane way around the issue.

Reply:
This is error workflow.